rust/psibase/src/table.rs

```diff
@@ -46,25 +47,24 @@ pub trait TableRecord: PackableOwned {
 
 pub trait Table<Record: TableRecord>: Sized {
     const TABLE_INDEX: u16;
-    const TABLE_SERVICE: AccountNumber;
     const SECONDARY_KEYS: u8;
 
-    fn new(db_id: DbId, prefix: Vec<u8>) -> Self;
+    fn with_prefix(db_id: DbId, prefix: Vec<u8>) -> Self;
     fn prefix(&self) -> &[u8];
     fn db_id(&self) -> DbId;
 
-    fn open() -> Self {
-        let prefix = (Self::TABLE_SERVICE, Self::TABLE_INDEX).to_key();
+    fn new() -> Self {
+        let prefix = (get_service(), Self::TABLE_INDEX).to_key();
         Self::create_table_from_prefix(prefix)
     }
 
-    fn open_custom(service: AccountNumber, table_index: u16) -> Self {
+    fn with_service(service: AccountNumber, table_index: u16) -> Self {
```

Comment:
remove `table_index`. This function handles any time service A reads service B's tables.

Reply:
What if service A, for convenience, also have the same table as table B under a different index? Eg. I could see services reusing the definition from `psispace-sys.ContentTable`. Should we always encourage duplicating the declaration?

Reply:
See https://doc-sys.psibase.io/rust-service/tables.html#storing-structs-defined-elsewhere

Reply:
I was working in the Global WebContent table and I realized that the consumer does not need to redeclare a shared struct keys like in the docs example.

Here's the WebContent global record struct: 
https://github.com/gofractally/psibase/pull/222/files#diff-64aad0707509e772a7c54f564cd388b06db283a4d6bb5042e9bcdd548a5d96cdR219

And here's how a simplified macro could be (to replace the following lines Table impl):
https://github.com/gofractally/psibase/pull/222/files#diff-ed73e7a1a9fad9c7b0753d5de5101331739bff714eef72fd1a054cbe72ba8e4cR75-R77

What do you think? Should I pursue this macro?